addon/-private/system/references/record.js

```diff
@@ -26,18 +26,25 @@ RecordReference.prototype.push = function(objectOrPromise) {
   });
 };
 
+RecordReference.prototype.isLoaded = function() {
+  return this.internalModel.isLoaded();
+};
+
 RecordReference.prototype.value = function() {
-  return this.internalModel.record;
+  if (this.isLoaded()) {
+    return this.internalModel.getRecord();
+  }
+
+  return null;
 };
 
 RecordReference.prototype.load = function() {
-  return this.store.findRecord(this.type, this._id);
+  return Ember.RSVP.resolve(this.store.findRecord(this.type, this._id));
```

Comment:
`Ember.RSVP.resolve(Ember.RSVP.reject("yolo")).catch(console.log.bind(console)) // => "yolo"`

But I added some tests to be sure :)